charm-slurmctld/src/charm.py

```diff
@@ -551,6 +556,56 @@ def _infludb_info_action(self, event):
         logger.debug(f"## InfluxDB-info action: {influxdb_info}")
         event.set_results({"influxdb": influxdb_info})
 
+    def _on_create_user_group(self, event):
+        """Create the user and group provided."""
+        user = self._user_group.user_name
+        user_uid = self._user_group.user_uid
+        group = self._user_group.group_name
+
+        # Create the group.
+        try:
+            subprocess.check_output(["groupadd", group])
```

Comment:
should we use the `user_id` as the group id too so we would `groupadd <group_name> --gid <user_uid>` ?

Reply:
> But what to do if the group is already taken? It will fail to create the group.

Yes. The status code returned will be 9 and then the code will log a warning telling the group already exists and nothing changes at all. The user will try to be created normally

Reply:
@heitorPB have a check please

Reply:
That makes sense. Let's keep that in mind for future work. But for now, let's get this merged!